### PR TITLE
Add max Nbr of Pages 

### PR DESCRIPTION
**Pull request by RobinDev**
New feature without breaking change, is it best to target 3.x or 2.x ?

This feature permit to limit the number of pages.

### CODE REVIEW

File: lib/Core/Pagerfanta.php [hunk 1]
```diff
@@ -546,4 +556,14 @@ private function needsToIntegerConversion($value): bool
     {
         return (\is_string($value) || \is_float($value)) && (int) $value == $value;
     }
+
+    /**
+     * @param int $maxNbPages
+     */
+    public function setMaxNbPages($maxNbPages): self
```
Comment (mbabker) [inline lib/Core/Pagerfanta.php]:
```suggestion
    public function setMaxNbPages(int $maxNbPages): self
```

It's a new method, we can add the typehint now.

File: lib/Core/Pagerfanta.php [hunk 2]
```diff
@@ -546,4 +556,14 @@ private function needsToIntegerConversion($value): bool
...
+    {
+        $this->maxNbPages = $maxNbPages;
```
Comment (mbabker) [inline lib/Core/Pagerfanta.php:568]: There should be a guard in this method to make sure we have a valid number here.  It looks like based on the implementation, an exception should be thrown if someone gives a number less than 0 here.

File: lib/Core/Pagerfanta.php [hunk 3]
```diff
@@ -377,12 +382,17 @@ private function notCachedNbResults(): bool
...
     public function getNbPages()
     {
+
         $nbPages = $this->calculateNbPages();
 
         if (0 === $nbPages) {
             return $this->minimumNbPages();
         }
 
+        if ($this->maxNbPages !== 0 && $this->maxNbPages < $nbPages) {
```
Comment (mbabker) [inline lib/Core/Pagerfanta.php]: Hmm, I'm not entirely sure if I want 0 to act as a "magic number" of sorts meaning there isn't a maximum number.  How about making the new property nullable (as well as the argument for the setter) and replace 0 with null here.

Then the guard in the setter would reject a number less than 1.

File: lib/Core/Pagerfanta.php (hunk 1 above)
Comment (RobinDev) [reply lib/Core/Pagerfanta.php]: Totally agree.

File: lib/Core/Pagerfanta.php (hunk 2 above)
Comment (RobinDev) [reply lib/Core/Pagerfanta.php:568]: Ok.

File: lib/Core/Pagerfanta.php (hunk 3 above)
Comment (RobinDev) [reply lib/Core/Pagerfanta.php]: No problem for that.

Having an internal 0 permit to typehint the property.
```
private int $maxNbPages = 0;
```

**Comment by RobinDev** [comment]
I don't know if it's the better way to do : I implement your suggestions and force pushed my commit.

**Pull request merged**